pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -171,8 +175,10 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 
 	// Fetch the Infinispan instance
 	infinispan := &infinispanv1.Infinispan{}
-	infinispan.Name = request.Name
-	infinispan.Namespace = request.Namespace
+	if result, err := kube.LookupResource(request.Name, request.Namespace, infinispan, r.client, reqLogger); result != nil {
```

Comment:
I think that's the reason why CI hangs, but need to be validated

Reply:
CI isn't hanging, it's just our tests are producing too much output for Travis so the job is being aborted. I have replaced `kube.LookupResource` with a simple `client.Get` so that we don't requeue when the Infinispan CR is not found .

Reply:
Why we need it. `err := r.update(infinispan, func() {` load initial Infinispan CR record.

Reply:
As we now watch for Secret updates, it's possible that a reconcile event will be fired when the the Infinispan CR no longer exists, causing `r.update` to fail as the object does not exist.

Reply:
This should be handles later with 
```
if errors.IsNotFound(err) {
			// Request object not found, could have been deleted after reconcile request.
			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
			// Return and don't requeue
			reqLogger.Info("Infinispan resource not found. Ignoring since object must be deleted")
			return reconcile.Result{}, nil
		}
```
We look at ConfigMap in the similar manner

Reply:
That's in the latest commit :slightly_smiling_face:

Reply:
The same code is placed after update. It should be handled all situation or nature a bit different
![image](https://user-images.githubusercontent.com/24249373/116223669-6f3a8980-a758-11eb-8993-9a61eb69a547.png)

Reply:
Argh yes you're correct. My addition isn't necessary. This was actually a hangover from when I was erroneously trying to use `EnqueueRequestForObject` to monitor the secrets and it was returning the secret name, so I added this to debug.